03-task-manager/starter/controllers/tasks.js

```diff
@@ -1,25 +1,72 @@
 const Task = require('../models/task');
-const getAllTasks = (req, res) => {
-    res.send('get all tasks');
-}
+const asyncWrapper = require('../middleware/async');
+const { createCustomError } = require('../errors/custom-error');
 
-const createTask = async(req, res) => {
+const getAllTasks = asyncWrapper( async (req, res) => {    
+    const tasks = await Task.find({});
+    res.status(200).json({tasks});
+    // res.status(200).json({tasks, amount: tasks.length});
+    // res.status(200).json({status: "success", data: {tasks,nbHits: tasks.length} });     
+})
+
+const createTask = asyncWrapper(async(req, res) => {   
     const task = await Task.create(req.body);
-    res.status(201).json({task});
-}
-const getTask = (req, res) => {
-    res.json({id:req.params.id});
-}
-const updateTask = (req, res) => {
-    res.send('update task');
-}
-const deleteTask = (req, res) => {
-    res.send('delete task');
-}
+    res.status(201).json({task});    
+})
+
+const getTask = asyncWrapper(async (req, res, next) => {   
+    const {id: taskID} = req.params;
+    const task = await Task.findOne({_id: taskID});
+    if(!task){
+       return next(createCustomError(`No task with id: ${taskID}`, 404));
+    }
+    res.status(200).json({task});   
+})
+
+const deleteTask = asyncWrapper( async (req, res) => {   
+    const {id: taskID} = req.params;
+    const task= await Task.findOneAndDelete({_id: taskID}); 
+    if(!task){
+        return next(createCustomError(`No task with id: ${taskID}`, 404));
+    }   
+    res.status(200).json({task});
+    // res.status(200).send();
+    // res.status(200).json({task:null, status: 'success' });
+    
+})
+const updateTask = asyncWrapper(async (req, res) => {    
+    const {id: taskID} = req.params;
+    const task = await Task.findOneAndUpdate({_id: taskID}, req.body, {
```

Comment:
Nitpick:
You'll also want to create a check here to determine if the `taskID` is a valid mongoose ID.